### PR TITLE
refactor: test class cleanups

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

Cleans up various test class usage across `tests/test_autonomy/test_cli`.

## Fixes

-

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: tests/test_autonomy/test_cli/base.py [hunk 1]
```diff
@@ -41,15 +41,15 @@ class BaseCliTest:
     cli_options: Tuple[str, ...]
 
     @classmethod
+    def setup_class(cls):
+        cls.cli_runner = CliRunner()
+        cls.cwd = Path.cwd().absolute()
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/base.py:46]: class level

File: tests/test_autonomy/test_cli/base.py [hunk 2]
```diff
@@ -41,15 +41,15 @@ class BaseCliTest:
...
-        cls,
+        self,
     ) -> None:
         """Setup test."""
-
-        cls.cli_runner = CliRunner()
-        cls.cwd = Path.cwd().absolute()
-        cls.t = Path(tempfile.mkdtemp())
-        cls.t.mkdir(exist_ok=True)
+        self.t = Path(tempfile.mkdtemp())
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/base.py:52]: before each test method

File: tests/test_autonomy/test_cli/base.py [hunk 3]
```diff
@@ -58,11 +58,10 @@ def run_cli(self, commands: Optional[Tuple[str, ...]] = None) -> Result:
 
         return self.cli_runner.invoke(cli=cli, args=(*self.cli_options, *commands))
 
-    @classmethod
     def teardown(
-        cls,
+        self,
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/base.py:62]: after each test method

File: tests/test_autonomy/test_cli/base.py [hunk 4]
```diff
@@ -41,15 +41,15 @@ class BaseCliTest:
     cli_options: Tuple[str, ...]
 
     @classmethod
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/base.py:43]: The `setup` and `teardown` methods are called before each `test_` method. The decorator makes no sense.

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_check_handlers.py [hunk 5]
```diff
@@ -117,13 +116,6 @@ def test_check_handlers_missing_handler(
                         [],
                     )
 
-    @classmethod
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_abci/test_check_handlers.py:120]: Already done by the parent

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py [hunk 6]
```diff
@@ -255,12 +253,6 @@ def test_failures(
             "Please provide path to specification file." in result.output
         ), result.output
 
-    @classmethod
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:258]: see above

File: tests/test_autonomy/test_cli/test_analyse/test_benchmarks.py [hunk 7]
```diff
@@ -62,31 +62,35 @@ class TestBenchmarks(BaseCliTest):
...
+        os.chdir(self.t)
 
-    def _run_test(self, block_type: str) -> None:
-        """Test run."""
+    def teardown(self) -> None:
+        """Teardown test method."""
+
+        super().teardown()
 
         with suppress(FileNotFoundError):
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_benchmarks.py:88]: Makes more sense to do in the teardown.

File: tests/test_autonomy/test_cli/test_analyse/test_benchmarks.py [hunk 8]
```diff
@@ -136,9 +140,3 @@ def test_only_total_blocks(
         """Test with only total blocks."""
 
         self._run_test(BlockTypes.TOTAL)
-
-    @classmethod
-    def teardown(cls) -> None:
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_benchmarks.py:141]: already done by the parent

File: tests/test_autonomy/test_cli/test_build_image.py [hunk 9]
```diff
@@ -70,10 +69,3 @@ def test_build_prod(
             len(self.docker_api.images(name=f"valory/oar-hello_world:{self.hash_}"))
             == 1
         )
-
-    @classmethod
-    def teardown(cls) -> None:
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_build_image.py:75]: see above

File: tests/test_autonomy/test_cli/test_deploy/test_build/test_deployment.py [hunk 10]
```diff
@@ -543,10 +542,3 @@ def test_build_dev_failures(
 
         assert result.exit_code == 1, result.output
         assert "Please provide proper value for --packages-dir" in result.output
-
-    @classmethod
-    def teardown(cls) -> None:
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_deploy/test_build/test_deployment.py:548]: see above

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 11]
```diff
@@ -67,72 +68,47 @@ def setup_class(cls) -> None:
...
         cls.author = get_default_author_from_cli_config()
 
-    @classmethod
     def setup(
-        cls,
+        self,
     ) -> None:
         """Setup test."""
 
-        cls._t = TemporaryDirectory()
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:76]: Most of this is already done in the base class!

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 12]
```diff
@@ -47,10 +47,11 @@
 fsm_specifications = list(VALORY_SKILLS_PATH.glob("**/fsm_specification.yaml"))
 
 
-class BaseScaffoldFSMTest(AEATestCaseEmpty):
+class BaseScaffoldFSMTest(AEATestCaseMany):
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:50]: @Karrenbelt the one you used already sets up an agent. Here we want to do this for each test method so we need to use the other class...

helps to read the code.

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 13]
```diff
@@ -67,72 +68,47 @@ def setup_class(cls) -> None:
...
-            cls.author,
+            self.author,
         )
 
+    def teardown(
+        self,
+    ) -> None:
+        """Teardown test."""
+        with suppress(OSError, FileExistsError, PermissionError):
+            shutil.rmtree(str(Path(self.t, self.agent_name)))
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:90]: remove the agent again

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 14]
```diff
@@ -67,72 +68,47 @@ def setup_class(cls) -> None:
...
+
     def scaffold_fsm(self, fsm_spec_file: Path) -> click.testing.Result:
         """Scaffold FSM."""
 
         *_, skill_name, _ = fsm_spec_file.parts
         skill_name = f"test_skill_{skill_name}"
         scaffold_args = ["scaffold", "fsm", skill_name, "--local", "--spec"]
         cli_args = [
-            "--registry-path",
-            self.packages_dir_path,
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:104]: if we use default class we don't need to do all this ...read the base classes!

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 15]
```diff
@@ -67,72 +68,47 @@ def setup_class(cls) -> None:
...
             *scaffold_args,
             fsm_spec_file,
         ]
         scaffold_result = self.run_cli_command(*cli_args, cwd=self.t / self.agent_name)
-        push_result = self.run_cli_command(
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:109]: no idea why this was needed here...

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 16]
```diff
@@ -167,13 +142,12 @@ def test_autonomy_test(self, fsm_spec_file: Path) -> None:
...
-        self.scaffold_fsm(fsm_spec_file)
+        scaffold_result = self.scaffold_fsm(fsm_spec_file)
+        assert scaffold_result.exit_code == 0
         cli_args = [
-            "--registry-path",
-            str(self.packages_dir_path),
             "test",
             "by-path",
-            str(self.t / "packages" / self.author / "skills" / skill_name),
+            str(self.t / self.agent_name / "skills" / skill_name),
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:150]: we want to run against the scaffolded skill, not the registry

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py (hunk 15 above)
Comment (DavidMinarsch) [reply tests/test_autonomy/test_cli/test_scaffold_fsm.py:109]: if we push to registry then we need to setup a new registry for each test..no need

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py (hunk 6 above)
Comment (Karrenbelt) [reply tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:258]: this is exactly one of the points i mentioned this morning I want to bring up in engineering sync you think people are aware of. They are not

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py (hunk 12 above)
Comment (Karrenbelt) [reply tests/test_autonomy/test_cli/test_scaffold_fsm.py:50]: so `Empty` is not empty :upside_down_face:

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py (hunk 15 above)
Comment (Karrenbelt) [reply tests/test_autonomy/test_cli/test_scaffold_fsm.py:109]: again added by Viraj. Already had reverted all of this on base branch

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py (hunk 14 above)
Comment (Karrenbelt) [reply tests/test_autonomy/test_cli/test_scaffold_fsm.py:104]: this is for you @angrybayblade

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py [hunk 17]
```diff
@@ -137,21 +136,18 @@ class TestCheckSpecs(BaseCliTest):
...
+        module_name = ".".join(self.skill_path.parts)
+        self.cls_name = ".".join([module_name, self.app_name])
 
-        cls.specification_path = cls.skill_path.parent / "fsm_specification.yaml"
-        shutil.copytree(ROOT_DIR / "packages", cls.t / "packages")
-        os.chdir(cls.t)
+        shutil.copytree(ROOT_DIR / "packages", self.t / "packages")
+        self.specification_path = (
+            self.t / self.skill_path.parent / "fsm_specification.yaml"
+        )
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:149]: point to the temp packages not the permanent one - risk of pollution

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py [hunk 18]
```diff
@@ -137,21 +136,18 @@ class TestCheckSpecs(BaseCliTest):
...
-    def setup(cls) -> None:
+    def setup(self) -> None:
         """Setup class."""
         super().setup()
 
-        module_name = ".".join(cls.skill_path.parts)
-        module = importlib.import_module(module_name)
-        cls.cls_name = ".".join([module_name, cls.app_name])
-
-        abci_app_cls = getattr(module, cls.app_name)
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:149]: some unused code

File: tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py [hunk 19]
```diff
@@ -205,8 +192,6 @@ def test_check_all(
         self,
     ) -> None:
         """Test --check-all flag."""
-
-        self._fix_corrupt_file()
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:209]: this is a very bad idea! We should never make assumptions about the order in which tests are executed. Each test needs to start from an appropriately clean environment.

Comment (Karrenbelt) [reply tests/test_autonomy/test_cli/test_analyse/test_abci/test_specs.py:209]: :+1:  yes, I had noticed this one too

**Pull request merged**